sklearn/semi_supervised/_self_training.py

```diff
@@ -0,0 +1,317 @@
+import numpy as np
+
+from ..base import BaseEstimator, clone
+from ..utils.validation import check_X_y, check_array, check_is_fitted
+from ..utils.metaestimators import if_delegate_has_method
+from ..utils import safe_mask
+from ..exceptions import ConvergenceWarning
+import warnings
+
+__all__ = ["SelfTrainingClassifier"]
+
+# Authors: Oliver Rausch   <rauscho@ethz.ch>
+#          Patrice Becker  <beckerp@ethz.ch>
+# License: BSD 3 clause
+
+
+def _validate_estimator(estimator):
+    """Make sure that an estimator implements the necessary methods."""
+    if estimator is None:
+        raise ValueError("base_classifier cannot be None!")
+
+    if not hasattr(estimator, "predict_proba"):
```

Comment:
This is only knowable for some estimators after fitting...

Reply:
I was not aware of this. I've moved the call to this method until after the estimator has been fitted for the first time.